compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -333,8 +333,12 @@ pub(super) fn type_of(tcx: TyCtxt<'_>, def_id: DefId) -> Ty<'_> {
                     find_opaque_ty_constraints_for_tait(tcx, def_id)
                 }
                 // Opaque types desugared from `impl Trait`.
-                ItemKind::OpaqueTy(OpaqueTy { origin: hir::OpaqueTyOrigin::FnReturn(owner) | hir::OpaqueTyOrigin::AsyncFn(owner), .. }) => {
-                    find_opaque_ty_constraints_for_rpit(tcx, def_id, owner)
+                ItemKind::OpaqueTy(OpaqueTy { origin: hir::OpaqueTyOrigin::FnReturn(owner) | hir::OpaqueTyOrigin::AsyncFn(owner), in_trait, .. }) => {
+                    if in_trait {
+                        span_bug!(item.span, "impl-trait in trait has no default")
```

Comment:
Actually, that's wrong if we supply a body to the trait function.

Reply:
We don't currently check that anyways -- I'm gonna leave that un-implemented in this PR I think.

Reply:
Specifically, I think we're gonna need new type-checking logic to check the body of a trait function with an RPITIT, so it'll need to be special cased in a way I haven't yet begun to think about.